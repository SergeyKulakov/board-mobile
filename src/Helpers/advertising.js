export const getDataWithAds = data => {
  const result = []

  data.forEach((el, key) => {
    const index = key + 1
    if (data[key + 1]) {
      if (index % 5 === 0) {
        result.push(el)
        result.push('ads')
      } else if (index === 3 || index % 7 === 0) {
        result.push(el)
        result.push('sponsor')
      } else result.push(el)
    } else result.push(el)
  })

  return result
}
