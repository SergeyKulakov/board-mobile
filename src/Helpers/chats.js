import moment from 'moment'

export const getMessageTime = date => {
  const TODAY = moment()
  const YESTERDAY = moment().date(TODAY.get('date') - 1)

  const data = Number(date)

  if (moment(data).isSame(TODAY, 'day')) return moment(data).format('HH:mm')
  if (moment(data).isSame(YESTERDAY, 'day')) return 'Yesterday'
  return moment(data).format('DD MMM YYYY')
}
