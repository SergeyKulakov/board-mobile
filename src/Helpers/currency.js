import data from 'Constants/country-currency-codes'

export const getSearchData = string => {
  if (!string || typeof string !== 'string') return data

  const value = string.toLowerCase()

  return data.filter(
    el =>
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.code.toLowerCase().indexOf(value) !== -1,
  )
}

export default data
