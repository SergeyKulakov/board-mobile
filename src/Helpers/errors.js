import _ from 'lodash'
import i18n from 'I18N'

export const getError = (error, path = 'apiErrors') => {
  const message = _.get(error, 'payload.message')
  const code = _.get(error, 'payload.code')

  if (
    error.payload === 'Internal server error' ||
    message === 'Internal server error' ||
    message === 'Internal Server Error' ||
    code === 'internal'
  )
    return i18n.t('apiErrors.InternalServerError')

  if (error.message === 'Unauthorized' || error.payload === 'Unauthorized') {
    return i18n.t('apiErrors.unauthorizedError')
  }

  if (!code) return `${message || error.payload}`

  return i18n.t(`${path}.${code}`)
}
