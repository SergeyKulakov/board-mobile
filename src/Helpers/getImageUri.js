import { BACKET } from 'react-native-dotenv'
import ProfileImagesManager from 'Services/Images/ProfileImagesManager'
import { profileImageTypes } from 'Services/Constants/profileImages.constants'
import _ from 'lodash'
import isUrl from 'is-url'

const s3 = 'https://s3.amazonaws.com'
const backet = BACKET
const serviceFolder = 'service-icons'
const jobsFolder = 'job-pics'

export const getServiceImage = fileName => {
  const iconName = fileName.replace(/\s/g, '%20')

  return `${s3}/${backet}/${serviceFolder}/${iconName}`
}

export const getJobImage = (id, fileName) =>
  `${s3}/${backet}/${jobsFolder}/${id}/${fileName}`

export const getProfileImage = ({ type, src, userId }) => {
  let source = src

  if (_.isString(src)) {
    if (/(data:image\/.*?;base64),/g.exec(src) || src.startsWith('http')) {
      source = src
    } else {
      source = ProfileImagesManager.getSignedURLForProfileImage(
        type,
        src,
        userId,
      )
    }
  } else if (_.isObject(src)) {
    source = null
  }

  return source
}

export const getProfileAvatar = (avatarURL, userId) => {
  if (isUrl(avatarURL)) return avatarURL

  if (_.isString(avatarURL)) {
    return getProfileImage({
      type: profileImageTypes.avatar,
      src: avatarURL,
      userId,
    })
  }

  return null
}
