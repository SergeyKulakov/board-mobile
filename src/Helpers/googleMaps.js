import _ from 'lodash'
import memoize from 'memoize-one'

const googleMaps = 'https://maps.googleapis.com/maps/api/staticmap'
const googleGeocoding = 'https://maps.googleapis.com/maps/api/geocode'
const params = {
  maptype: 'roadmap',
  size: '600x300',
  zoom: '15',
  markers: [
    'color:blue|label:S|40.702147,-74.015794', // example
  ],
  key: 'AIzaSyAOrI9daMiZzf1t2oPscaFD4FpPod4yTR8',
  center: 'BrooklynBridge,New York,NY', // example
}

export const getMapImage = (geolocation = {}) => {
  const { lat, lon } = geolocation
  if (lat && lon) {
    return `${googleMaps}?center=${lat},${lon}&zoom=${params.zoom}&size=${
      params.size
    }&maptype=${
      params.maptype
    }&markers=color:red%7Clabel:S%7C${lat},${lon}&key=${params.key}`
  }

  return `https://maps.googleapis.com/maps/api/staticmap?center=BrooklynBridge,New%20York,NY&zoom=0&size=600x300&maptype=roadmap&key=${
    params.key
  }`
}

export const getLocatinByGeocoding = (lat, lon) => {
  return `${googleGeocoding}/json?latlng=${lat},${lon}&key=${params.key}`
}

export const getGeocodeObg = memoize(geocode => {
  if (_.isString(geocode) && !_.isEmpty(geocode)) {
    const geodata = geocode.split('/')
    return {
      lat: Number(geodata[0]),
      lon: Number(geodata[1]),
    }
  }

  return {}
})
