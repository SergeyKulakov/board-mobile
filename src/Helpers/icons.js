import colors from 'Themes/Colors'
import _ from 'lodash'

export const getNavigationIconStyle = (icon, allSize = 20) => {
  const defaultProps = {
    size: allSize,
    color: colors.blue,
  }

  if (_.isString(icon) || !icon) {
    return {
      name: icon,
      ...defaultProps,
    }
  }

  return {
    type: icon.type,
    name: icon.name,
    color: icon.color || defaultProps.color,
    size: icon.size || defaultProps.size,
  }
}
