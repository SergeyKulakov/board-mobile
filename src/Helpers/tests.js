export const getTestID = value => `[testID="${value}"]`

export const serverErrorObg = {
  error: {
    payload: {
      message: 'message',
      code: 'code',
    },
  },
}
