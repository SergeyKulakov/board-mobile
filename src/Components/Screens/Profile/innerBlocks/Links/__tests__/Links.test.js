import React from 'react'
import { shallow } from 'enzyme'
import { getTestID } from 'Helpers/tests'

import 'jest-styled-components'

import { Links as Component } from '../index'

describe('Profile.Links', () => {
  const defaultProps = {
    text: {
      videoLinksLengthError: 'videoLinksLengthError',
      websiteLinksLengthErrorFree: 'websiteLinksLengthErrorFree',
      websiteLinksLengthErrorPremium: 'websiteLinksLengthErrorPremium',
    },
    values: {
      videoLinks: [],
      websiteLinks: [],
    },
    setFieldValue: () => null,
    toast: () => null,
    isPremium: false,
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('add video links', () => {
    describe('successful', () => {
      it('without premium & without other video links', () => {
        const props = {
          ...defaultProps,
          setFieldValue: jest.fn(),
          toast: jest.fn(),
        }

        const wrapper = shallow(<Component {...props} />)

        const key = 'videoLinks'
        const args = {
          link: 'https://youtobe.com',
          image: '83270491823u01uifr0iuh0eiv',
          title: 'title',
          description: 'description',
        }

        wrapper
          .find(getTestID('videoLinks'))
          .props()
          .onAdd(args)

        expect(props.toast).not.toHaveBeenCalled()
        expect(props.setFieldValue).toHaveBeenCalledTimes(1)
        expect(props.setFieldValue).toHaveBeenCalledWith(
          key,
          props.values.videoLinks.concat([
            {
              id: props.values.videoLinks.length + 1,
              link: args.link,
              image: args.image,
              info: args.title,
            },
          ]),
        )
      })
      it('with premium', () => {
        const props = {
          ...defaultProps,
          setFieldValue: jest.fn(),
          toast: jest.fn(),
          values: {
            videoLinks: [
              {
                id: 1,
                link: 'https://google.com',
                image: '14102389470398r1039r3',
                info: 'title',
              },
            ],
          },
          isPremium: true,
        }

        const wrapper = shallow(<Component {...props} />)

        const key = 'videoLinks'
        const args = {
          link: 'https://youtobe.com',
          image: '83270491823u01uifr0iuh0eiv',
          title: 'title',
          description: 'description',
        }

        wrapper
          .find(getTestID('videoLinks'))
          .props()
          .onAdd(args)

        expect(props.toast).not.toHaveBeenCalled()
        expect(props.setFieldValue).toHaveBeenCalledTimes(1)
        expect(props.setFieldValue).toHaveBeenCalledWith(
          key,
          props.values.videoLinks.concat([
            {
              id: props.values.videoLinks.length + 1,
              link: args.link,
              image: args.image,
              info: args.title,
            },
          ]),
        )
      })
    })

    describe('failure add video link [permissions]', () => {
      it('without premium', () => {
        const props = {
          ...defaultProps,
          setFieldValue: jest.fn(),
          toast: jest.fn(),
          values: {
            videoLinks: [
              {
                id: 1,
                link: 'https://google.com',
                image: '14102389470398r1039r3',
                info: 'title',
              },
            ],
          },
          isPremium: false,
        }

        const wrapper = shallow(<Component {...props} />)

        const args = {
          link: 'https://youtobe.com',
          image: '83270491823u01uifr0iuh0eiv',
          title: 'title',
          description: 'description',
        }

        wrapper
          .find(getTestID('videoLinks'))
          .props()
          .onAdd(args)

        expect(props.setFieldValue).not.toHaveBeenCalled()
        expect(props.toast).toHaveBeenCalledTimes(1)
        expect(props.toast).toHaveBeenCalledWith(
          props.text.videoLinksLengthError,
        )
      })
    })
  })

  describe('add website links', () => {
    describe('successful', () => {
      it('without premium', () => {
        const props = {
          ...defaultProps,
          setFieldValue: jest.fn(),
          toast: jest.fn(),
          isPremium: false,
          values: {
            ...defaultProps.values,
            websiteLinks: [],
          },
        }

        const wrapper = shallow(<Component {...props} />)

        const key = 'websiteLinks'
        const args = {
          link: 'https://youtobe.com',
          image: '83270491823u01uifr0iuh0eiv',
          title: 'title',
          description: 'description',
        }

        wrapper
          .find(getTestID('websiteLinks'))
          .props()
          .onAdd(args)

        expect(props.setFieldValue).toHaveBeenCalledTimes(1)
        expect(props.setFieldValue).toHaveBeenCalledWith(key, [
          {
            id: 1,
            link: args.link,
            image: args.image,
            info: args.title,
          },
        ])
      })

      it('with premium', () => {
        const props = {
          ...defaultProps,
          setFieldValue: jest.fn(),
          toast: jest.fn(),
          isPremium: true,
          values: {
            ...defaultProps.websiteLinks,
            websiteLinks: [
              {
                id: 1,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
              {
                id: 2,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
              {
                id: 3,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
            ],
          },
        }

        const wrapper = shallow(<Component {...props} />)

        const key = 'websiteLinks'
        const args = {
          link: 'https://youtobe.com',
          image: '83270491823u01uifr0iuh0eiv',
          title: 'title',
          description: 'description',
        }

        wrapper
          .find(getTestID('websiteLinks'))
          .props()
          .onAdd(args)

        expect(props.setFieldValue).toHaveBeenCalledTimes(1)
        expect(props.setFieldValue).toHaveBeenCalledWith(
          key,
          props.values.websiteLinks.concat([
            {
              id: 4,
              link: args.link,
              image: args.image,
              info: args.title,
            },
          ]),
        )
      })
    })

    describe('failure', () => {
      it('without premium', () => {
        const props = {
          ...defaultProps,
          setFieldValue: jest.fn(),
          toast: jest.fn(),
          isPremium: false,
          values: {
            websiteLinks: [
              {
                id: 1,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
            ],
          },
        }

        const wrapper = shallow(<Component {...props} />)

        const args = {
          link: 'https://youtobe.com',
          image: '83270491823u01uifr0iuh0eiv',
          title: 'title',
          description: 'description',
        }

        wrapper
          .find(getTestID('websiteLinks'))
          .props()
          .onAdd(args)

        expect(props.setFieldValue).not.toHaveBeenCalled()
        expect(props.toast).toHaveBeenCalledTimes(1)
        expect(props.toast).toHaveBeenCalledWith(
          props.text.websiteLinksLengthErrorFree,
        )
      })

      it('with premium', () => {
        const props = {
          ...defaultProps,
          setFieldValue: jest.fn(),
          toast: jest.fn(),
          isPremium: true,
          values: {
            ...defaultProps.values,
            websiteLinks: [
              {
                id: 1,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
              {
                id: 2,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
              {
                id: 3,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
              {
                id: 4,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
              {
                id: 5,
                link: 'https://youtobe.com',
                image: '83270491823u01uifr0iuh0eiv',
                info: 'description',
              },
            ],
          },
        }

        const wrapper = shallow(<Component {...props} />)

        const args = {
          link: 'https://youtobe.com',
          image: '83270491823u01uifr0iuh0eiv',
          title: 'title',
          description: 'description',
        }

        wrapper
          .find(getTestID('websiteLinks'))
          .props()
          .onAdd(args)

        expect(props.setFieldValue).not.toHaveBeenCalled()
        expect(props.toast).toHaveBeenCalledTimes(1)
        expect(props.toast).toHaveBeenCalledWith(
          props.text.websiteLinksLengthErrorPremium,
        )
      })
    })
  })
  it('delete link', () => {
    const props = {
      ...defaultProps,
      setFieldValue: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.setFieldValue).not.toHaveBeenCalled()

    for (let i = 0; i < wrapper.find('LinkPreview').length; i++) {
      wrapper
        .find('LinkPreview')
        .get(i)
        .props.onDelete({})
    }

    expect(props.setFieldValue).toHaveBeenCalledTimes(2)
  })
})
