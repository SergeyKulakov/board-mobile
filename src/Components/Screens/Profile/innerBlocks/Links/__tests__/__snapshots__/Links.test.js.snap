// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.Links render snapshot 1`] = `
<Fragment>
  <ShadowBox>
    <LinkPreview
      data={Array []}
      inputType="video"
      onAdd={[Function]}
      onDelete={[Function]}
      testID="videoLinks"
      toast={[Function]}
    />
  </ShadowBox>
  <ShadowBox>
    <LinkPreview
      data={Array []}
      disabledLoading={true}
      onAdd={[Function]}
      onDelete={[Function]}
      testID="websiteLinks"
      toast={[Function]}
    />
  </ShadowBox>
</Fragment>
`;
